src/oauth-provider.ts

```diff
@@ -1622,6 +1678,13 @@ class OAuthProviderImpl {
     grantData.refreshTokenId = newRefreshTokenId;
     grantData.refreshTokenWrappedKey = newRefreshTokenWrappedKey;
 
+    // Update refresh token expiration if a new TTL was provided by callback, otherwise keep existing expiration
+    if (refreshTokenTTLUpdated) {
+      // If TTL is 0 or null, remove expiration; otherwise set new expiration
```

Comment:
Treating `null` as "no expiration" is OK, but treating 0 as "no expiration" seems dangerous. If anything it should be treated as "expires immediately" (though I'm also OK with it meaning "don't issue a refresh token").